scripts/logbook.py

```diff
@@ -198,6 +701,23 @@ def write_record(content,dest_file):
     body = {'valueInputOption' : "USER_ENTERED", 'data' : data}
     result = service.spreadsheets().values().batchUpdate(spreadsheetId=spreadsheetId, body=body).execute()
 
+# Fetch UDF details
+def get_details(record_instrument,pro):
+    udf_detail=[]
+    if record_instrument["details"] != []:
```

Comment:
Now this check is not required, as python will not iterate on empty list anyway. So I would re-write the function something like

```
def get_details(record_instrument,pro):
    udf_detail = []
    for item in record_instrument.get("details", []):
        try:
             udf_detail.append(item+":"+pro.udf[item])
        except KeyError, e:
             continue
    return ",".join(udf_detail) if udf_detail else "-"
```